src/daemon/daemon.cpp

```diff
@@ -1277,39 +1270,32 @@ try // clang-format on
 {
     mpl::ClientLogger<RecoverReply> logger{mpl::level_from(request->verbosity_level()), *config->logger, server};
 
-    fmt::memory_buffer errors;
-    std::vector<decltype(deleted_instances)::key_type> instances_to_recover;
-    for (const auto& name : request->instance_names().instance_name())
+    const auto instances_and_status =
+            find_requested_instances(request->instance_names().instance_name(), deleted_instances,
+                                     std::bind(&Daemon::check_instance_exists, this, std::placeholders::_1));
+    const auto& instances = instances_and_status.first; // use structured bindings instead in C++17
+    const auto& status = instances_and_status.second;   // idem
+
+    if(status.ok())
     {
-        auto it = deleted_instances.find(name);
-        if (it == deleted_instances.end())
+        for (const auto& name : instances)
         {
-            it = vm_instances.find(name);
-            if (it == vm_instances.end()) // noop otherwise
-                fmt::format_to(errors, "instance \"{}\" does not exist\n", name);
+            auto it = deleted_instances.find(name);
+            if(it != std::end(deleted_instances))
+            {
+                it->second->shutdown(); // not sure why this is needed, but leaving it alone
```

Comment:
I imagine this is a copy & pasta error from back when:)  I think it's safe to remove.

Reply:
Yes it looks like that. Alright, I will remove it.